src/openassetio-python/tests/package/hostApi/test_manager.py

```diff
@@ -654,6 +675,51 @@ def call_callbacks(*_args):
             mock.ANY,  # error
         )
 
+    def test_pager_kept_alive_by_retaining_shared_ptr(
+        self,
+        manager,
+        a_ref,
+        mock_manager_interface,
+        a_host_session,
+        a_batch_element_error,
+        an_empty_traitsdata,
+        an_entity_trait_set,
+        a_context,
+        invoke_getWithRelationshipPaged_success_cb,
+        invoke_getWithRelationshipPaged_error_cb,
+    ):
+        # pylint: disable=too-many-locals
+
+        two_refs = [a_ref, a_ref]
+        page_size = 3
+
+        error_callback = mock.Mock()
+
+        method = mock_manager_interface.mock.getWithRelationshipPaged
+
+        def call_callbacks(*_args):
+            invoke_getWithRelationshipPaged_success_cb(0, DyingPagerInterface())
+            invoke_getWithRelationshipPaged_error_cb(1, a_batch_element_error)
+
+        method.side_effect = call_callbacks
+
+        pagers = []
+        manager.getWithRelationshipPaged(
+            two_refs,
+            an_empty_traitsdata,
+            page_size,
+            a_context,
+            lambda _idx, pager: pagers.append(pager),
+            error_callback,
+            resultTraitSet=an_entity_trait_set,
+        )
+
+        """
+        Without PyRetainingSharedPtr, will raise
+        "Tried to call pure virtual function "EntityReferencePagerInterface::get"
+        """
```

Comment:
[consider] We generally use comments vs string literals. Same below.

Reply:
Switched in c341414d